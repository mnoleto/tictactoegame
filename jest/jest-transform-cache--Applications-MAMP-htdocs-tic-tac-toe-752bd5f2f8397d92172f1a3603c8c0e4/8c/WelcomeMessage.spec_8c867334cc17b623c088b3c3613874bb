/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_WelcomeMessage = require('./WelcomeMessage'); /* istanbul ignore next */var _WelcomeMessage2 = _interopRequireDefault(_WelcomeMessage);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// WelcomeMessage is responsible for render the game grid
// CONTRACTS
// 1. 

describe('WelcomeMessage', function () {
  var props = /* istanbul ignore next */void 0,mountedWelcomeMessage = /* istanbul ignore next */void 0;
  var welcomeMessage = function welcomeMessage() {
    if (!mountedWelcomeMessage) {
      mountedWelcomeMessage = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_WelcomeMessage2.default, props));

    }
    return mountedWelcomeMessage;
  };

  beforeEach(function () {
    props = {};
    mountedWelcomeMessage = undefined;
  });

  it('render the WelcomeMessage', function () {
    var welcomeMessageWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_WelcomeMessage2.default, /* istanbul ignore next */null));
    expect(welcomeMessageWrapper).toMatchSnapshot();
  });

  it('render a div as root element', function () {
    var welcomeMessageElement = welcomeMessage().find('div');
    var wrappingDiv = welcomeMessageElement.first();
    expect(wrappingDiv.children()).toEqual(welcomeMessage().children());
  });
});