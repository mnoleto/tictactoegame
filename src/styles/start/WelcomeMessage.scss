.welcome-message {
  margin-bottom: 30px;
  padding: 0 20%;
  text-align: center;
  width: 100%;
}