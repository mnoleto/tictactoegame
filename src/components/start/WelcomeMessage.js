import React from 'react';
import '../../styles/start/WelcomeMessage.scss';

// WelcomeMessage is responsible for render the game welcome message for the first page
const WelcomeMessage = () => {
  return (
    <div className="welcome-message">
      <p>Tic-tac-toe (also known as noughts and crosses or Xs and Os) is a paper-and-pencil game for two players, X and O, who take turns marking the spaces in a 3Ã—3 grid. The player who succeeds in placing three of their marks in a horizontal, vertical, or diagonal row wins the game.</p>
    </div>
  );
};

export default WelcomeMessage;