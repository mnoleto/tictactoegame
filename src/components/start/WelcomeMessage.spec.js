import React from 'react';
import { mount, shallow } from 'enzyme';
import WelcomeMessage from './WelcomeMessage';

// WelcomeMessage is responsible for render the game grid
// CONTRACTS
// 1. 

describe('WelcomeMessage', () => {
  let props, mountedWelcomeMessage;
  const welcomeMessage = () => {
    if(!mountedWelcomeMessage) {
      mountedWelcomeMessage = mount(
        <WelcomeMessage {...props} />
      );
    }
    return mountedWelcomeMessage;
  };

  beforeEach(() => {
    props = {};
    mountedWelcomeMessage = undefined;
  });

  it('render the WelcomeMessage', () => {
    const welcomeMessageWrapper = shallow(<WelcomeMessage />);
    expect(welcomeMessageWrapper).toMatchSnapshot();
  });

  it('render a div as root element', () => {
    const welcomeMessageElement = welcomeMessage().find('div');
    let wrappingDiv = welcomeMessageElement.first();
    expect(wrappingDiv.children()).toEqual(welcomeMessage().children());
  });

  it('render a "p" as root element', () => {
    const pElement = welcomeMessage().find('p');
    expect(pElement.length).toEqual(1);
  });
});