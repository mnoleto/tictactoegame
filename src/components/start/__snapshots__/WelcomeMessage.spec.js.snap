exports[`WelcomeMessage render the WelcomeMessage 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="welcome-message">
    <p>
        Tic-tac-toe (also known as noughts and crosses or Xs and Os) is a paper-and-pencil game for two players, X and O, who take turns marking the spaces in a 3×3 grid. The player who succeeds in placing three of their marks in a horizontal, vertical, or diagonal row wins the game.
    </p>
</div>,
  "nodes": Array [
    <div
      className="welcome-message">
      <p>
            Tic-tac-toe (also known as noughts and crosses or Xs and Os) is a paper-and-pencil game for two players, X and O, who take turns marking the spaces in a 3×3 grid. The player who succeeds in placing three of their marks in a horizontal, vertical, or diagonal row wins the game.
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <WelcomeMessage />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="welcome-message">
          <p>
                    Tic-tac-toe (also known as noughts and crosses or Xs and Os) is a paper-and-pencil game for two players, X and O, who take turns marking the spaces in a 3×3 grid. The player who succeeds in placing three of their marks in a horizontal, vertical, or diagonal row wins the game.
          </p>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="welcome-message">
          <p>
                    Tic-tac-toe (also known as noughts and crosses or Xs and Os) is a paper-and-pencil game for two players, X and O, who take turns marking the spaces in a 3×3 grid. The player who succeeds in placing three of their marks in a horizontal, vertical, or diagonal row wins the game.
          </p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <WelcomeMessage />,
}
`;
